### PR TITLE
Using Github App token to trigger CI for version increment PRs

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Using Github App token to trigger CI on PRs created for version increment from the version.yml workflow. (https://github.com/opensearch-project/common-utils/pull/111#issuecomment-1039635451)
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/version.yml [hunk 1]
```diff
@@ -11,6 +11,14 @@ jobs:
     env:
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
+       - name: GitHub App token
+        id: github_app_token
+        uses: tibdex/github-app-token@v1.5.0
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+          installation_id: 22958780
```
Comment (owaiskazi19) [inline .github/workflows/version.yml:20]: Do we have any significance of this id?

Comment (VachaShah) [reply .github/workflows/version.yml:20]: Its the installation id of the `opensearch-trigger-bot` Github App.

Comment (owaiskazi19) [reply .github/workflows/version.yml:20]: Is it fine to expose it or should we add it in secrets?

Comment (VachaShah) [reply .github/workflows/version.yml:20]: Its fine. Its the same across all repos, we have been using it in [backport](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/backport.yml#L22) and [dependabot](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/dependabot_pr.yml#L18) workflows.

Comment (owaiskazi19) [reply .github/workflows/version.yml:20]: Sounds good. Thanks for answering the questions.

**Review by owaiskazi19 — APPROVED** [review]
LGTM!

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
> Looks like `1.x` and `1.2` don't have the `version.yml` file. @dblock Do we want to backport it?

I think we should backport it to 1.x, or that workflow won't run when we branch and then release 1.3.0. Would you please do that? Thank you.

**Comment by VachaShah** [comment]
> > Looks like `1.x` and `1.2` don't have the `version.yml` file. @dblock Do we want to backport it?
> 
> I think we should backport it to 1.x, or that workflow won't run when we branch and then release 1.3.0. Would you please do that? Thank you.

Sure, I added a `backport 1.x` label to the original PR.

**Pull request merged**